### PR TITLE
Move away from Travis

### PR DESCRIPTION
**Pull request by xuhdev**
We use GitHub Actions for everything other than pypy. We use GitLab CI for pypy tests, which GitHub actions take about 20min to finish. We hereby drop both Travis and AppVeyor, as GitHub Actions has a better concurrency, and the fact that MacOS test takes 6 min won't affect too much to our workflow.

### CODE REVIEW

**Comment by xuhdev** [comment]
I don't know why the gitlab tests show my icon, while other people seem to have the gitlab icon, but I think it's fine.

Do you also receive the test failure notification from GitHub Actions? If not, I think we can merge now and adjust notification a bit after moving the repo just in case of URL changes.

After you approve, I'll change required tests, disable Travis and AppVeyor, and merge this PR. Then we should be good to move back to CODAIT.

**Comment by edwardleardi** [comment]
> I don't know why the gitlab tests show my icon, while other people seem to have the gitlab icon, but I think it's fine.

I also see your icon for gitlab, not a big deal though

> Do you also receive the test failure notification from GitHub Actions? If not, I think we can merge now and adjust notification a bit after moving the repo just in case of URL changes.

I haven't received any GH Actions notifications yet

> After you approve, I'll change required tests, disable Travis and AppVeyor, and merge this PR. Then we should be good to move back to CODAIT.

sounds good

**Comment by edwardleardi** [comment]
GitHub Actions logs don't have colorized output 😢

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#
```
Comment (edwardleardi) [inline .gitlab-ci.yml:1]: How does the GitLab repo stay up to date if we are pushing upstream to GitHub?

**Review by edwardleardi — APPROVED** [review]
LGTM, just one question

File: .gitlab-ci.yml (hunk 1 above)
Comment (xuhdev) [reply .gitlab-ci.yml:1]: It is a [pull mirror](https://docs.gitlab.com/ee/user/project/repository/repository_mirroring.html). We don't hack this, GitLab CI is supposed to be used like this for GitHub repos (automatic mirror + CI): https://docs.gitlab.com/ee/ci/ci_cd_for_external_repos/github_integration.html

**Pull request merged**